tests/test_scalars.py

```diff
@@ -339,147 +381,219 @@ def test_timedelta() -> None:
     check(assert_type(td / 3.5, pd.Timedelta), pd.Timedelta)
-    check(assert_type(td / np.array([1, 2, 3]), np.ndarray), np.ndarray)
-    check(assert_type(td / np.array([1.2, 2.2, 3.4]), np.ndarray), np.ndarray)
-    check(assert_type(td / pd.Series([1, 2, 3]), pd.Series), pd.Series)
-    check(assert_type(td / pd.Series([1.2, 2.2, 3.4]), pd.Series), pd.Series)
-    check(assert_type(td / pd.Index([1, 2, 3]), pd.TimedeltaIndex), pd.TimedeltaIndex)
     check(
-        assert_type(td / pd.Index([1.2, 2.2, 3.4]), pd.TimedeltaIndex),
-        pd.TimedeltaIndex,
+        assert_type(td / np.array([1, 2, 3]), npt.NDArray[np.timedelta64]), np.ndarray
     )
+    check(
+        assert_type(td / np.array([1.2, 2.2, 3.4]), npt.NDArray[np.timedelta64]),
+        np.ndarray,
+    )
+    check(assert_type(td / pd.Series([1, 2, 3]), pd.Series), pd.Series)
+    check(assert_type(td / pd.Series([1.2, 2.2, 3.4]), pd.Series), pd.Series)
+    check(assert_type(td / i_idx, pd.TimedeltaIndex), pd.TimedeltaIndex)
+    check(assert_type(td / f_idx, pd.TimedeltaIndex), pd.TimedeltaIndex)
+
+    # Note: None of the reverse truediv work
+    # check(assert_type(3 / td, pd.Timedelta), pd.Timedelta)
+    # check(assert_type(3.5 / td, pd.Timedelta), pd.Timedelta)
+    # check(assert_type(np.array([1, 2, 3]) / td, npt.NDArray[np.timedelta64]), np.ndarray)
+    # check(assert_type(np.array([1.2, 2.2, 3.4]) / td, npt.NDArray[np.timedelta64]),np.ndarray,)
+    # check(assert_type(pd.Series([1, 2, 3]) / td, pd.Series), pd.Series)
+    # check(assert_type(pd.Series([1.2, 2.2, 3.4]) / td, pd.Series), pd.Series)
+    # check(assert_type(i_idx / td, pd.TimedeltaIndex), pd.TimedeltaIndex)
+    # check(assert_type(f_idx / td, pd.TimedeltaIndex), pd.TimedeltaIndex)
 
     check(assert_type(td % 3, pd.Timedelta), pd.Timedelta)
     check(assert_type(td % 3.5, pd.Timedelta), pd.Timedelta)
     check(assert_type(td % td, pd.Timedelta), pd.Timedelta)
-    check(assert_type(td % np.array([1, 2, 3]), np.ndarray), np.ndarray)
-    check(assert_type(td % np.array([1.2, 2.2, 3.4]), np.ndarray), np.ndarray)
+    check(
+        assert_type(td % np.array([1, 2, 3]), npt.NDArray[np.timedelta64]), np.ndarray
+    )
+    check(
+        assert_type(td % np.array([1.2, 2.2, 3.4]), npt.NDArray[np.timedelta64]),
+        np.ndarray,
+    )
     check(assert_type(td % pd.Series([1, 2, 3]), pd.Series), pd.Series)
     check(assert_type(td % pd.Series([1.2, 2.2, 3.4]), pd.Series), pd.Series)
-    check(assert_type(td % pd.Index([1, 2, 3]), pd.TimedeltaIndex), pd.TimedeltaIndex)
+    check(assert_type(td % i_idx, pd.TimedeltaIndex), pd.TimedeltaIndex)
+
     check(
-        assert_type(td % pd.Index([1.2, 2.2, 3.4]), pd.TimedeltaIndex),
+        assert_type(td % f_idx, pd.TimedeltaIndex),
         pd.TimedeltaIndex,
     )
 
-    check(assert_type(abs(td), pd.Timedelta), pd.Timedelta)
+    check(assert_type(td.__abs__(), pd.Timedelta), pd.Timedelta)
```

Comment:
I don't think we should call the "hidden" operator in the tests.  Why can't you use `abs(td)` here?